src/features/orders/ordersActions.ts

```diff
@@ -277,6 +277,19 @@ export const approve =
         approveAmount
       );
 
+      if (isAppError(tx)) {
+        const appError = tx;
+        dispatch(setErrors([appError]));
+
+        if (appError.error && "message" in appError.error) {
+          dispatch(declineTransaction(appError.error.message));
+        }
+
+        dispatch(setStatus("failed"));
```

Comment:
```suggestion
        dispatch(setStatus("failed"));
        handleOrderError(dispatch, tx);
```

This does the same basically.

Reply:
Great. Had just copied this over from the function below, will update.